### ISSUE TITLE
[Pipeline] TokenFlow: Consistent Diffusion Features for Consistent Video Editing

### ISSUE THREAD

**Issue by clarencechen** [huggingface/diffusers#4892]
### Model/Pipeline/Scheduler description

TokenFlow is a framework that enables consistent video editing, using a pre-trained text-to-image diffusion model, without any further training or finetuning via nearest-neighbor matching of (noised) latent diffusion features between frames.

The generative AI revolution has been recently expanded to videos. Nevertheless, current state-of-the-art video models are still lagging behind image models in terms of visual quality and user control over the generated content. In this work, we present a framework that harnesses the power of a text-to-image diffusion model for the task of text-driven video editing. Specifically, given a source video and a target text-prompt, our method generates a high-quality video that adheres to the target text, while preserving the spatial layout and dynamics of the input video. Our method is based on our key observation that consistency in the edited video can be obtained by enforcing consistency in the diffusion feature space. We achieve this by explicitly propagating diffusion features based on inter-frame correspondences, readily available in the model. Thus, our framework does not require any training or fine-tuning, and can work in conjunction with any off-the-shelf text-to-image editing method. We demonstrate state-of-the-art editing results on a variety of real-world videos.

For more see the [project webpage](https://diffusion-tokenflow.github.io/).

### Open source status

- [X] The model implementation is available
- [X] The model weights are available (Only relevant if addition is not a scheduler).

### Provide useful links for the implementation

Code: https://github.com/omerbt/TokenFlow
Paper: https://diffusion-tokenflow.github.io/TokenFlow_Arxiv.pdf
Relevant Authors: @omerbt

**Comment by patrickvonplaten** [comment]
I wonder if it makes sense to port this pipeline as it's already quite simple to use with the original code: https://github.com/omerbt/TokenFlow 

It has a pip install package and automatic loading and it already uses all the `diffusers` speed-ups out of the box as far as I can see.

We could maybe just add a doc page and link to the GitHub repo and only port it if the GitHub repo becomes stale?

**Comment by sayakpaul** [comment]
Agree with @patrickvonplaten here. Going forward, I think we should follow that approach. This helps us unburden ourselves from the maintenance a bit and helps promote the original works of the authors (as they're built on top of `diffusers` and come with easy usability in mind).
